File: java-webflux/src/main/java/com/ederfmatos/javawebflux/repository/FixedTransactionRepository.java
package com.ederfmatos.javawebflux.repository;

import com.ederfmatos.javawebflux.entity.Transaction;
import com.ederfmatos.javawebflux.enums.Currency;
import static java.math.BigDecimal.TEN;
import java.time.Duration;
import java.time.ZonedDateTime;
import lombok.RequiredArgsConstructor;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

@Repository
@RequiredArgsConstructor
@ConditionalOnProperty(value = "repository.type", havingValue = "fixed", matchIfMissing = true)
public class FixedTransactionRepository implements TransactionRepository {
    @Override
    public Mono<Void> create(Transaction transaction) {
        return Mono.delay(Duration.ofMillis(20))
                .then();
    }

    @Override
    public Mono<Transaction> findById(String id) {
        return Mono.delay(Duration.ofMillis(20))
                .map(value -> new Transaction(id, "description", TEN, TEN, Currency.BRL, ZonedDateTime.now()));
    }
}

File: kotlin-webflux/src/main/kotlin/com/ederfmatos/kotlinwebflux/repository/FixedTransactionRepository.kt
package com.ederfmatos.kotlinwebflux.repository

import com.ederfmatos.kotlinwebflux.entity.Transaction
import com.ederfmatos.kotlinwebflux.enums.Currency
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono
import reactor.core.scheduler.Schedulers
import java.math.BigDecimal.TEN
import java.time.Duration
import java.time.ZonedDateTime

@Repository
@ConditionalOnProperty(value = ["repository.type"], havingValue = "fixed", matchIfMissing = true)
class FixedTransactionRepository : TransactionRepository {

    override fun create(transaction: Transaction): Mono<Void> {
        return Mono.delay(Duration.ofMillis(20))
            .subscribeOn(Schedulers.boundedElastic())
            .flatMap { Mono.empty() }
    }

    override fun findById(id: String): Mono<Transaction> {
        return Mono.delay(Duration.ofMillis(20))
            .subscribeOn(Schedulers.boundedElastic())
            .map { Transaction(id, "description", Currency.BRL, TEN, TEN, ZonedDateTime.now()) }
    }
}